lib/decoder/real-time.js

```diff
@@ -0,0 +1,66 @@
+const url = require('url')
+
+// https://github.com/PRX/Infrastructure/blob/master/cdn/dovetail3-cdn.yml#L289
+const FIELDS = [
+  'timestamp',
+  'c-ip',
+  'sc-status',
+  'cs-method',
+  'cs-uri-stem',
+  'cs-user-agent',
+  'cs-referer',
+  'x-forwarded-for',
+  'sc-content-len',
+  'sc-range-start',
+  'sc-range-end',
+]
+const ASCII = /^[\x00-\xFF]*$/
+
+/**
+ * check for cloudfront real-time log events, with known TSV fields
+ */
+exports.detect = function (buffer) {
+  const str = buffer.toString('utf-8')
+  return ASCII.test(str) && str.split('\t').length === FIELDS.length
+}
+
+/**
+ * parse tab separated log lines
+ */
+exports.decode = async function (buffer) {
+  const parts = buffer.toString('utf-8').trim().split('\t')
+  const data = {}
+  FIELDS.forEach((name, index) => {
+    if (parts[index] === '-') {
```

Comment:
If `-` is a placeholder character in the fields, I'd add a comment to make more obvious.

Reply:
It's a CW log convention, which I don't see really documented anywhere.  So that's nice.  +1 - will add a const.